source/include/core_mqtt_agent.h

```diff
@@ -58,7 +58,14 @@ typedef enum MQTTCommandType
     DISCONNECT,  /**< @brief Call MQTT_Disconnect(). */
     TERMINATE,   /**< @brief Exit the command loop and stop processing commands. */
     NUM_COMMANDS /**< @brief The number of command types handled by the agent. */
-} MQTTAgentCommandType_t;
+};
+
+/**
+ * @ingroup mqtt_agent_enum_types
+ * @brief use with values of enum MQTTCommandType
+ * to set the command type in struct MQTTAgentCommand
+ */
+typedef uint32_t MQTTAgentCommandType_t;
```

Comment:
This removes any 'hints' that an IDE gets from the type of the variable. When writing code, the smart suggestions that IDEs provide seem important to me.

Secondly, why a `uint32_t`? Why not a `uin8_t`? I am curious.

Reply:
I hadn't considered IDE auto-completions. I'll revert and try casting as Archit suggested.